spec/Section 6 -- Execution.md

```diff
@@ -360,9 +445,71 @@ either resolving to {null} if allowed or further propagated to a parent field.
 If this occurs, any sibling fields which have not yet executed or have not yet
 yielded a value may be cancelled to avoid unnecessary work.
 
+Additionally, async payload records in {subsequentPayloads} must be filtered if
+their path points to a location that has resolved to {null} due to propagation
+of a field error. This is described in
+[Filter Subsequent Payloads](#sec-Filter-Subsequent-Payloads). These async
+payload records must be removed from {subsequentPayloads} and their result must
+not be sent to clients. If these async records have not yet executed or have not
+yet yielded a value they may also be cancelled to avoid unnecessary work.
+
 Note: See [Handling Field Errors](#sec-Handling-Field-Errors) for more about
 this behavior.
 
+### Filter Subsequent Payloads
+
+When a field error is raised, there may be async payload records in
+{subsequentPayloads} with a path that points to a location that has been removed
+or set to null due to null propagation. These async payload records must be
+removed from subsequent payloads and their results must not be sent to clients.
+
+In {FilterSubsequentPayloads}, {nullPath} is the path which has resolved to null
```

Comment:
what would be the expected behavior if the asyncRecord is under a null inside a list?
With a query like

```graphql
{
  products {
    id
     ... @defer {
       name
     }
   }
 }
 ```
 
 If we get as first payload:
 
 ```json
{
  "products": [
    null,
    {
      "id": 1
    }
  ]
 }
 ```
 
 we would expect the second payload to be:
 
 ```json
 {
  "incremental":[
    {
      "path": [
        "products",
        0
      ],
      "data": null
    },
    {
      "path": [
        "products",
        1
      ],
      "data": {
        "name": "chair"
      }
    }
  ]
 }
 ```
 
 But if the payload was:
 
 ```json
{
  "products": [
    null,
    null
  ]
 }
 ```
 
 Would we cancel the second payload or return:
 
 ```json
 {
  "incremental":[
    {
      "path": [
        "products",
        0
      ],
      "data": null
    },
    {
      "path": [
        "products",
        1
      ],
      "data": null
    }
  ]
 }
```